sound/soc/sof/sof-audio.c

```diff
@@ -433,6 +433,26 @@ struct snd_sof_dai *snd_sof_find_dai(struct snd_soc_component *scomp,
 	return NULL;
 }
 
+/*
+ * Helper to get SSP MCLK from a pcm_runtime.
+ * Return 0 if not exist.
+ */
+int snd_sof_dai_get_ssp_mclk(struct snd_soc_pcm_runtime *rtd)
+{
+	struct snd_soc_component *component =
+		snd_soc_rtdcom_lookup(rtd, SOF_AUDIO_PCM_DRV_NAME);
+	struct snd_sof_dai *dai =
+		snd_sof_find_dai(component, (char *)rtd->dai_link->name);
+	int clk_freq = 0;
+
+	/* use the tplg configured mclk if existed */
+	if (dai && dai->dai_config && dai->dai_config->type == SOF_DAI_INTEL_SSP)
+		clk_freq = dai->dai_config->ssp.mclk_rate;
```

Comment:
how about just `return dai->dai_config->ssp.mclk_rate;` here and `return 0;` below?